### PR TITLE
benchmark user properties update

### PR DESCRIPTION
**Pull request by tiina303**
- Add new update properties function
- nit
- nit
- Sanitizing propertiesOnce here as it's public fn
- fix event props getting changed
- review comments
- Code instead of postgres function + making sure we lock the person for update during fetch + test for matching key
- benchmarking person updates

## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

## How did you test this code?

*Please describe.*

### CODE REVIEW

File: plugin-server/src/utils/db/db.ts [hunk 1]
```diff
@@ -540,6 +541,41 @@ export class DB {
         return client ? kafkaMessages : updatedPerson
     }
 
+    public async updatePersonPropertiesOld(
```
Comment (tiina303) [inline plugin-server/src/utils/db/db.ts:544]: copied from process-event.ts to make the comparison over the whole function for the old way too

**Comment by tiina303** [comment]
Not planning to merge this, but to verify benchmarks

Ran with `jest --runInBand --forceExit --verbose false`
```
      │ (index) │ control │  new  │ diffMs │ diffMagnitude │
      ├─────────┼─────────┼───────┼────────┼───────────────┤
      │   10    │  6.76   │ 10.67 │  3.91  │     1.58      │
      │   20    │  7.19   │ 9.92  │  2.73  │     1.38      │
      │   50    │  6.45   │ 10.76 │  4.31  │     1.67      │
      │   100   │  7.23   │ 11.73 │  4.5   │     1.62      |
```

**Comment by yakkomajuri** [comment]
Unsure what you ran to get those big numbers. Here's from my side:

```
      ┌─────────┬─────────┬──────┬────────┬───────────────┐
      │ (index) │ control │ new  │ diffMs │ diffMagnitude │
      ├─────────┼─────────┼──────┼────────┼───────────────┤
      │   10    │  1.45   │ 1.71 │  0.25  │     1.17      │
      │   20    │  0.91   │ 1.26 │  0.35  │     1.39      │
      │   50    │  1.19   │ 1.97 │  0.79  │     1.66      │
      │   100   │  1.06   │ 1.93 │  0.87  │     1.82      │
      └─────────┴─────────┴──────┴────────┴───────────────┘
 ```

**Review by yakkomajuri — APPROVED** [review]
results are good, happy to proceed

**Comment by tiina303** [comment]
I don't really want to merge it in the current state because of copied code in `updatePersonPropertiesOld`, could update to change the function or could just not land this

**Pull request closed without merging**